main.nf

```diff
@@ -1131,112 +1131,119 @@ process busco_download_db {
-        cp run_${assembly}/short_summary_${assembly}.txt short_summary_${assembly}.txt
-
-        for f in run_${assembly}/single_copy_busco_sequences/*faa; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*faa >${assembly}_buscos.faa || touch ${assembly}_buscos.faa
-            break
-        done
-        for f in run_${assembly}/single_copy_busco_sequences/*fna; do 
-            [ -e "\$f" ] && cat run_${assembly}/single_copy_busco_sequences/*fna >${assembly}_buscos.fna || touch ${assembly}_buscos.fna
-            break
-        done
-        """
-    }
+    # place db in extra folder to ensure BUSCO recognizes it as path (instead of downloading it)
+    mkdir dataset
+    mv ${db} dataset/
+
+    busco \
+        --mode genome \
+        --in ${bin} \
+        --lineage_dataset dataset/${db} \
+        --config \${config_file} \
+        --cpu "${task.cpus}" \
+        --out "BUSCO" > ${bin}_busco.log
+
+    cp BUSCO/short_summary.specific.${db}.BUSCO.txt short_summary.specific.${db}.${bin}.txt
+
+    for f in BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*faa; do
+        [ -e "\$f" ] && cat BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*faa >${bin}_buscos.faa
+        break
+    done
+    for f in BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*fna; do
+        [ -e "\$f" ] && cat BUSCO/run_${db}/busco_sequences/single_copy_busco_sequences/*fna >${bin}_buscos.fna
+        break
+    done
+    """
 }
 
+// preprare channels for downstream processes
+ch_busco_multiqc = ch_busco_multiqc.map{it[2]}
+ch_busco_to_summary = ch_busco_to_summary.map{it[2]}
 
-process busco_plot { 
-    publishDir "${params.outdir}/GenomeBinning/QC/", mode: 'copy'
+// group by assembler and sample for plotting
+ch_busco_plot = ch_busco_plot.groupTuple(by: [0,1])
+
+process busco_plot {
+    tag "$assembler-$sample"
+    publishDir "${params.outdir}/GenomeBinning/QC/BUSCO/", mode: 'copy'
 
     input:
-    file(summaries) from busco_summary_to_plot.collect()
-    val(assemblersample) from busco_assembler_sample_to_plot.collect()
+    set val(assembler), val(sample), file(summaries) from ch_busco_plot
 
     output:
-    file("*busco_figure.png")
-    file("BUSCO/*busco_figure.R")
-    file("BUSCO/*busco_summary.txt")
-    file("busco_summary.txt") into busco_summary
+    file("${assembler}-${sample}-busco_figure.png")
+    file("${assembler}-${sample}-busco_figure.R")
+    file("${assembler}-${sample}-busco_summary.txt")
 
     script:
-    def assemblersampleunique = assemblersample.unique()
+    def name = "${assembler}-${sample}"
     """
-    #for each assembler and sample:
-    assemblersample=\$(echo \"$assemblersampleunique\" | sed 's/[][]//g')
-    IFS=', ' read -r -a assemblersamples <<< \"\$assemblersample\"
+    # replace dots in bin names within summary file names with underscores
+    # (currently, v4.1.2, generate_plot.py does not allow further dots)
```

Comment:
Its v4.0.6, isnt it?

Reply:
true (just also holds for newer version), will adjust